### PR TITLE
Initial plugin component setup

### PR DESCRIPTION
**Pull request by silvano-a**
This PR is part of the initial base for this plugin. 
It is not complete and as of now does not adhere to any standards. this will be changed in feature PRs

### CODE REVIEW

File: src/Licon.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/**
+ * Licon constructor
```
Comment (TimGroenenberg) [inline src/Licon.js]: Add `.`

File: src/Licon.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        var instanceSettings = $.extend(this.settings, settings);
+        if (instanceSettings.targetUrl === false) {
+            return;
+        }
+        this.instances[this.counter] = new this.Loader(this, instanceSettings);
+        this.instances[this.counter].initialize();
+        this.counter ++;
+    }
+    return this;
+};
```
Comment (TimGroenenberg) [inline src/Licon.js]: Add newline.

File: src/components/Components.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Registers the prototype for adapters
```
Comment (TimGroenenberg) [inline src/components/Components.js]: Add `.`

File: src/Licon.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    this.counter = 0;
+}
+
+/**
+ * Handles jQuery calls.
+ * 
+ * @param {*|jQuery|HTMLElement} $object
+ * @param {Object} settings
+ */
+Licon.prototype.handleCall = function($object, settings)
```
Comment (TimGroenenberg) [inline src/Licon.js]: Shouldn't there be a space between function and the arguments?

File: src/components/content/Append.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
+/**
+ * Adapter which functions as Ajax RequestManager.
+ *
+ * @param loader
+ * @param settings
+ * @constructor
```
Comment (TimGroenenberg) [inline src/components/content/Append.js]: Add newline above.

File: src/components/navigation/Scrolltrigger.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
+/**
+ * Adapter which
```
Comment (TimGroenenberg) [inline src/components/navigation/Scrolltrigger.js]: Uncompleted comment.

File: src/components/navigation/Scrolltrigger.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    this.contentFinished = false;
+
+    this.$trigger = $('<div/>',
+        {
+            text: 'Meer laden',
+            class: 'licon-js-scrolltrigger',
+            'data-licon-id' : this.settings.$element.attr('data-licon-id')
+        });
+
+
```
Comment (TimGroenenberg) [inline src/components/navigation/Scrolltrigger.js]: Remove newline.

File: src/components/navigation/Scrolltrigger.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+{
+    this.active = true;
+    this.$renderedNavigation = $(e.navigation);
+    $(e.navigation).on('click', $.proxy(this.clickEventHandler, this));
+    $(window).on('scroll.Licon', $.proxy(this.scrollEventHandler, this))
+};
+
+Licon.prototype.Components.prototype.Scrolltrigger.prototype.clickEventHandler = function (e)
+{
+    console.log('click!');
```
Comment (TimGroenenberg) [inline src/components/navigation/Scrolltrigger.js]: Remove console log.

File: src/components/navigation/Scrolltrigger.js [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+Licon.prototype.Components.prototype.Scrolltrigger.prototype.getNavigation = function ()
+{
+    return {
+        'navigationLocation': this.navigationLocation,
+        '$navigation': this.$trigger
+    }
+};
+
+
+Licon.prototype.Components.prototype.Scrolltrigger.prototype.stageLoaded = function (e)
```
Comment (TimGroenenberg) [inline src/components/navigation/Scrolltrigger.js]: Add docblock.

File: src/components/navigation/Scrolltrigger.js [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+ * @returns {{navigationLocation: *, $navigation: (*|jQuery|HTMLElement)}}
+ */
+Licon.prototype.Components.prototype.Scrolltrigger.prototype.getNavigation = function ()
+{
+    return {
+        'navigationLocation': this.navigationLocation,
+        '$navigation': this.$trigger
+    }
+};
+
```
Comment (TimGroenenberg) [inline src/components/navigation/Scrolltrigger.js]: Remove newline.

File: src/components/navigation/Scrolltrigger.js [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+Licon.prototype.Components.prototype.Scrolltrigger.prototype.stageLoaded = function (e)
+{
+    this.active = true;
+    this.$renderedNavigation = $(e.navigation);
+    $(e.navigation).on('click', $.proxy(this.clickEventHandler, this));
+    $(window).on('scroll.Licon', $.proxy(this.scrollEventHandler, this))
+};
+
+Licon.prototype.Components.prototype.Scrolltrigger.prototype.clickEventHandler = function (e)
```
Comment (TimGroenenberg) [inline src/components/navigation/Scrolltrigger.js]: Add docblock.

File: src/components/navigation/Scrolltrigger.js [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+{
+    console.log('click!');
+    if(this.active === false)
```
Comment (TimGroenenberg) [inline src/components/navigation/Scrolltrigger.js]: Add space after `if`.

File: src/components/navigation/Scrolltrigger.js [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+    {
+        return;
+    }
+    this.active = false;
```
Comment (TimGroenenberg) [inline src/components/navigation/Scrolltrigger.js]: Add newline.

File: src/components/navigation/Scrolltrigger.js [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+    this.checkForNextPage();
+};
+
+Licon.prototype.Components.prototype.Scrolltrigger.prototype.scrollEventHandler = function (e)
+{
+    console.log(this);
```
Comment (TimGroenenberg) [inline src/components/navigation/Scrolltrigger.js]: Remove console log.

File: src/components/navigation/Scrolltrigger.js [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+    if(this.active === false) {
```
Comment (TimGroenenberg) [inline src/components/navigation/Scrolltrigger.js]: Add space after `if`.

File: src/components/navigation/Scrolltrigger.js [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+
+    $loader = e.$renderedNavigation;
+
+    if ($loader.length == 0 || this.loading == true) {
+        return;
+    }
+
+    var documentTop = $(window).scrollTop();
+    var documentBottom = documentTop + $(window).height();
+
```
Comment (TimGroenenberg) [inline src/components/navigation/Scrolltrigger.js]: Remove newline.

File: src/components/navigation/Scrolltrigger.js [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+    if ((elementTop <= documentBottom) && (elementTop >= documentTop) && this.loading == false) {
+        this.active = false;
+        this.checkForNextPage();
+    }
+};
+
+Licon.prototype.Components.prototype.Scrolltrigger.prototype.checkForNextPage = function ()
+{
+    console.log('yo');
```
Comment (TimGroenenberg) [inline src/components/navigation/Scrolltrigger.js]: Remove console log.

File: src/components/navigation/Scrolltrigger.js [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+
+    var elementTop = $loader.offset().top;
+
+    if ((elementTop <= documentBottom) && (elementTop >= documentTop) && this.loading == false) {
+        this.active = false;
+        this.checkForNextPage();
+    }
+};
+
+Licon.prototype.Components.prototype.Scrolltrigger.prototype.checkForNextPage = function ()
```
Comment (TimGroenenberg) [inline src/components/navigation/Scrolltrigger.js]: Add docblock.

File: src/components/navigation/Scrolltrigger.js [hunk 19]
```diff
@@ -0,0 +1,91 @@
...
+    console.log('click!');
+    if(this.active === false)
+    {
+        return;
+    }
+    this.active = false;
+    this.checkForNextPage();
+};
+
+Licon.prototype.Components.prototype.Scrolltrigger.prototype.scrollEventHandler = function (e)
```
Comment (TimGroenenberg) [inline src/components/navigation/Scrolltrigger.js]: Add docblock.

File: src/components/navigation/Scrolltrigger.js [hunk 20]
```diff
@@ -0,0 +1,91 @@
...
+    if ((elementTop <= documentBottom) && (elementTop >= documentTop) && this.loading == false) {
+        this.active = false;
+        this.checkForNextPage();
+    }
+};
+
+Licon.prototype.Components.prototype.Scrolltrigger.prototype.checkForNextPage = function ()
+{
+    console.log('yo');
+    if(this.contentFinished === false)
```
Comment (TimGroenenberg) [inline src/components/navigation/Scrolltrigger.js]: Add space after `if`.

File: src/components/navigation/Scrolltrigger.js [hunk 21]
```diff
@@ -0,0 +1,91 @@
...
+    {
+        this.Loader.setRequestData(this.pageData);
+    }
+};
```
Comment (TimGroenenberg) [inline src/components/navigation/Scrolltrigger.js]: Add newline.

File: src/components/request/Ajax.js [hunk 22]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * Sends the data back to the loader which prompted the load.
+ *
+ * @param data
+ */
+Licon.prototype.Components.prototype.Ajax.prototype.handleRequestComplete = function (data)
+{
+    this.Loader.RequestManagerCallback(data);
+};
```
Comment (TimGroenenberg) [inline src/components/request/Ajax.js]: Add newline.

File: src/jQuery.Licon.js [hunk 23]
```diff
@@ -0,0 +1,35 @@
+(function ($)
+{
+    var settings = {
```
Comment (TimGroenenberg) [inline src/jQuery.Licon.js]: Is this needed?

File: src/jQuery.Licon.js [hunk 24]
```diff
@@ -0,0 +1,35 @@
...
+        }
+
+        if (licon instanceof Licon === false) {
+            licon = new Licon(settings);
+        }
+
+        test = licon.handleCall(this, settings);
+
+        //Return this for jQuery chaining
+        return test;
```
Comment (TimGroenenberg) [inline src/jQuery.Licon.js]: Do we want this in the project?

File: src/jQuery.Licon.js [hunk 25]
```diff
@@ -0,0 +1,35 @@
...
+ include order
+
+ <script src="/liconjs/src/Licon.js"></script>
+ <script src="/liconjs/src/adapters/Adapters.js"></script>
+ <script src="/liconjs/src/adapters/request/Ajax.js"></script>
+ <script src="/liconjs/src/adapters/content/Replace.js"></script>
+
+ <script src="/liconjs/src/loader/Loader.js"></script>
+ <script src="/liconjs/src/jQuery.Licon.js"></script>
+ */
```
Comment (TimGroenenberg) [inline src/jQuery.Licon.js]: Add newline.

File: src/loader/Loader.js [hunk 26]
```diff
@@ -0,0 +1,71 @@
...
+};
+
+/**
+ * Initializes all components and registers adapters.
+ */
+Licon.prototype.Loader.prototype.initialize = function ()
+{
+    this.Components = new this.Application.Components();
+
+    console.log(this.Components);
```
Comment (TimGroenenberg) [inline src/loader/Loader.js]: Remove console log.

File: src/loader/Loader.js [hunk 27]
```diff
@@ -0,0 +1,71 @@
...
+/**
+ * Callback for when the Request manager is done with the request.
+ *
+ * @param data
+ */
+Licon.prototype.Loader.prototype.RequestManagerCallback = function (data)
+{
+    this.components['ContentManager'].showContent(data);
+};
+
```
Comment (TimGroenenberg) [inline src/loader/Loader.js]: Remove newline.

File: src/jQuery.Licon.js (hunk 23 above)
Comment (silvano-a) [reply src/jQuery.Licon.js]: this is currently used to cast all settings to an object if the user supplies the argument in an array for example.

File: src/jQuery.Licon.js (hunk 24 above)
Comment (silvano-a) [reply src/jQuery.Licon.js]: nope!

**Pull request merged**